robot/EDA/resources/EDA.py

```diff
@@ -418,3 +436,33 @@ def verify_dropdown_field_status(self, **kwargs):
             expected_value = bool(expected_value == "disabled")
             if not str(expected_value).lower() == str(actual_value).lower() :
                 raise Exception (f"Drop down field {field} status is {actual_value} instead of {expected_value}")
+
+    def verify_checkbox_value(self,**kwargs):
+        """ This method validates the checkbox value for the field passed in kwargs
+            Pass the field name and expected value to be verified from the tests using
+            keyword arguments. true - checked, false - unchecked
+        """
+        self.selenium.execute_javascript("window.scrollTo(0, document.body.scrollHeight)")
+        self.selenium.execute_javascript("window.scrollTo(document.body.scrollHeight, 0)")
+        for field,expected_value in kwargs.items():
+            locator = eda_lex_locators["eda_settings_system"]["default_checkbox"].format(field)
+            self.selenium.page_should_contain_element(locator)
+            self.selenium.wait_until_element_is_visible(locator,
```

Comment:
cant u directly use wait_until_element_is_visible directly? if the locator is not there then it would not be visible

Reply:
That's how it was implemented at first but then while testing I saw inconsistent failures due to wait so I had to add this step to avoid inconsistent failures